### ISSUE TITLE
about averaging gradient in Semantic segmentation

### ISSUE THREAD

**Issue by ascust** [dmlc/gluon-cv#591]
I found the crossentropy loss is basically a wrapper of nd array softmaxoutput operator. In semantic segmentation, the loss's default normalization is "valid". So I assume the gradient is already normalized with regard to the number of valid points. So in optimizer step, why do we have to average it again using the batch size as in:
```
self.optimizer.step(self.args.batch_size)
```
I am also wondering how this is related when it comes to multiple gpus, where gradients from different gpus are supposed to be averaged. Please help me clear this up.

**Comment by ascust** [comment]
@zhanghang1989 @zhreshold @piiswrong

**Comment by zhanghang1989** [comment]
We usually normalize the gradients per-pixel per-sample (per-gpu, per-mini-batch). We usually adjust the learning rate accordingly.

**Comment by ascust** [comment]
@zhanghang1989 so I guess in training
```
self.optimizer.step(self.args.batch_size)
```
the gradient should be rescaled by the number of gpus instead of the batch size. Am I right?
